src/ring_theory/valuation/valuation_subring.lean

```diff
@@ -435,4 +435,64 @@ unit_group_order_embedding.strict_mono
 
 end unit_group
 
+section nonunits
+
+/-- The nonunits of a valuation subring, as a subsemigroup of `K`-/
+def nonunits : subsemigroup K :=
```

Comment:
Is this worthy of a module docstring?

Reply:
I don't know... it's just like most other definitions in mathlib. It's fine if you want to add one, but I think the docstring is also fine.